tests/modules/aclcheck.c

```diff
@@ -51,6 +51,52 @@ int set_aclcheck_key(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
     return REDISMODULE_OK;
 }
 
+/* A wrap for SET command with ACL check on the key. */
+int set_aclcheck_prefixkey(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+    if (argc < 4) {
+        return RedisModule_WrongArity(ctx);
+    }
+
+    int permissions;
+    const char *flags = RedisModule_StringPtrLen(argv[1], NULL);
+
+    if (!strcasecmp(flags, "W")) {
+        permissions = REDISMODULE_CMD_KEY_UPDATE;
+    } else if (!strcasecmp(flags, "R")) {
+        permissions = REDISMODULE_CMD_KEY_ACCESS;
+    } else if (!strcasecmp(flags, "*")) {
+        permissions = REDISMODULE_CMD_KEY_UPDATE | REDISMODULE_CMD_KEY_ACCESS;
+    } else if (!strcasecmp(flags, "~")) {
+        permissions = 0; /* Requires either read or write */
+    } else {
+        RedisModule_ReplyWithError(ctx, "INVALID FLAGS");
+        return REDISMODULE_OK;
+    }
+
+    /* Check that the key can be accessed */
+    RedisModuleString *user_name = RedisModule_GetCurrentUserName(ctx);
+    RedisModuleUser *user = RedisModule_GetModuleUserFromUserName(user_name);
```

Comment:
I believe this might be the common use-case.

100% of the time, we have a context at hand, but this API requires a user. It would have been a better UX for the clients of this function if it actually accepted a context instead of the user object, which everyone will have to obtain manually using the corresponding APIs, while the user object is already set within the context. I understand that it is quite extendable and future-proof to have it accept the `RedisModuleUser` instead, but perhaps we should just refactor it into two functions, one of which calls another, also extracting the user from the context? This will be much simpler and won't require performing exactly the same changes in all the modules.

Reply:
I don't have a strong opinion between using a user or context, except to the fact that the `RedisModule_ACLCheckKeyPermissions` also takes a RedisModuleUser and we better be aligned.